crypto/getenv.c

```diff
@@ -15,6 +15,50 @@
 #include "internal/cryptlib.h"
 #include "internal/e_os.h"
 
+/*
+ * Additional checks for securely allowing access to selected environment variables
+ */
+#if defined (OPENSSL_NETCAP_ALLOW_ENV) && defined(__linux__)
+
+/*
+ * Utility function to check if a string is in an array of strings. returns true if found
+ */
+int str_in_list(const char *candidate, const char *list[], int n)
+{
+  if ( (candidate!=NULL) && (list!=NULL))
+  {
+    int len = n;
+    if (len>=1)
+    {
+        int i;
+        for ( i=0; i<len; i++)
+        {
+            if (strcmp(candidate,list[i])==0)
+              // found
+              return 1;
+        }
+    }
+  }
+  // drop-through case -- not found (or invalid parameters)
+  return 0;
+}
+
+char *restricted_getenv(const char *name)
+{
+    /* Environment variables to permit in secure mode (subject to conditions). Extend as needed */
+    const char * permitted_env [ ] = { "OPENSSL_MODULES", "OPENSSL_CONF" };
+    if(name!=NULL && (!OPENSSL_issetugid()))
+    {
+      if (str_in_list(name,permitted_env,sizeof(permitted_env)/sizeof(char *)))
```

Comment:
Rather than hard-code the type (`char *`), please use something like

```
#define OSSL_NELEM(x) (sizeof(x)/sizeof((x)[0]))
```

Reply:
The `OSSL_NELEM` macro is defined in `internal/nelem.h`.

Reply:
Thanks for the tip - have refactored as per suggestion
(I've left explicit conditional checks in at this stage)